tribits/ci_support/CDashAnalyzeReportRandomFailures.py

```diff
@@ -0,0 +1,339 @@
+    # and start of scanning summaries and table
+    cdashReportData.htmlEmailBodyTop +="</p>\n\n<p>\n"
+
+    # B.2) Get each nonpassing test's testing history
+    for nonpassingTest in initialNonpassingTestsLOD:
+
+      # Remove unique jenkins run ID from build name
+      correctedBuildName = self.extractBuildNameStrategy.getCoreBuildName(nonpassingTest['buildName'])
+
+      testNameBuildName = nonpassingTest['testname']+"_"+nonpassingTest['buildName']
+      testNameBuildName = CDQAR.getCompressedFileNameIfTooLong(testNameBuildName)
+
+      print("\n Getting history from "+dateRangeStr+" for\n"+\
+            "  Test name: "+nonpassingTest['testname']+"\n"+\
+            "  Build name: "+correctedBuildName)
+
+      groupNameNormUrl, = CDQAR.normalizeUrlStrings(groupName)
+
+      testHistoryFilters = \
+        "filtercount=3&showfilters=1&filtercombine=and"+\
+        "&field1=testname&compare1=63&value1="+nonpassingTest['testname']+\
+        "&field2=groupname&compare2=63&value2="+groupNameNormUrl+\
+        "&field3=buildname&compare3=63&value3="+correctedBuildName
+
+      testHistoryQueryFilters = dateUrlField+"&"+testHistoryFilters
+
+      testHistoryCacheFile = testHistoryCacheDir+"/"+\
+        CDQAR.getCompressedFileNameIfTooLong(testNameBuildName+".json", ext=".json")
+
+      print("\n  Creating file to write test history:\n   "+testHistoryCacheFile)
+
+      testHistoryQueryUrl = CDQAR.getCDashQueryTestsQueryUrl(
+        cdashSiteUrl, cdashProjectName, None, testHistoryQueryFilters)
+      testHistoryBrowserUrl = CDQAR.getCDashQueryTestsBrowserUrl(
+        cdashSiteUrl, cdashProjectName, None, testHistoryQueryFilters)
+
+      testHistoryLOD = CDQAR.downloadTestsOffCDashQueryTestsAndFlatten(
+        testHistoryQueryUrl, testHistoryCacheFile, alwaysUseCacheFileIfExists=True)
+
+      print("\n  Size of test history: "+str(len(testHistoryLOD)))
+
+      if printUrlMode == 'all':
+        print("\n  CDash test history browser URL for "+nonpassingTest['testname']+" "+\
+          correctedBuildName+":\n\n"+"   "+testHistoryBrowserUrl)
+        print("\n  CDash test history query URL for "+nonpassingTest['testname']+" "+\
+          correctedBuildName+":\n\n"+"   "+testHistoryQueryUrl)
+
+      if len(testHistoryLOD) < 2:
+        print("\n  Size of test history too small for any comparisons, skipping ...\n")
+        continue
+
+      # B.3) Split full testing history to passed and nonpassed lists of dicts
+      passingTestHistoryLOD = [test for test in testHistoryLOD if test.get("status") == "Passed"]
+      nonpassingTestHistoryLOD = [test for test in testHistoryLOD if test.get("status") == "Failed"]
+      nonpassingSha1Pairs = set()
+
+      print("\n  Num of passing tests in test history: "+str(len(passingTestHistoryLOD)))
+      print("\n  Num of nonpassing tests in test history: "+str(len(nonpassingTestHistoryLOD)))
+
+      buildSummaryCacheDir = testQueriesCacheDir+"/build_summary_cache/"+testNameBuildName
```

Comment:
Instead of calling `getCompressedFileNameIfTooLong()` at line 107 above, why not just call it here?   But I guess if  the variable `testNameBuildName` is only used to create the file name above and this directory name, then I guess there is no reason not to compress the name above at line 107.